prune.go

```diff
@@ -232,30 +289,8 @@ func txnsPruneC(txnsName string) string {
 // TODO(mjs) - this knows way too much about mgo/txn's internals and
 // with a bit of luck something like this will one day be part of
 // mgo/txn.
-func PruneTxns(db *mgo.Database, txns *mgo.Collection) error {
-	workingSetName := txns.Name + ".prunetemp"
-	workingSet := db.C(workingSetName)
-	defer workingSet.DropCollection()
-
-	// Load the ids of all completed and aborted txns into a separate
-	// temporary collection.
-	logger.Debugf("loading all completed transactions")
-	pipe := txns.Pipe([]bson.M{
-		// This used to use $in but that's much slower than $gte.
-		{"$match": bson.M{"s": bson.M{"$gte": taborted}}},
-		{"$project": bson.M{"_id": 1}},
-		{"$out": workingSetName},
-	})
-	pipe.Batch(maxBatchDocs)
-	pipe.AllowDiskUse()
-	if err := pipe.All(&bson.D{}); err != nil {
-		return fmt.Errorf("reading completed txns: %v", err)
-	}
-
-	count, err := workingSet.Count()
-	if err != nil {
-		return fmt.Errorf("getting txn count: %v", err)
-	}
+func PruneTxns(db *mgo.Database, oracle Oracle, txns *mgo.Collection) error {
```

Comment:
It feels better (to me) to declare that it is depending on access to these things, rather than secreting access from another object.